os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -106,7 +106,7 @@ static int allocateregions(FAR struct elf_loadinfo_s *loadinfo)
 	}
 
 	loadinfo->binp->ramstart = *allocs[0] = (uintptr_t)kmm_memalign(sizes[0], totalsize);
-	if (loadinfo->binp->ramstart == NULL) {
+	if (!loadinfo->binp->ramstart) {
```

Comment:
`kmm_memalign` return the pointer or NULL, and the meaning of ramstart is a address even its type is uint.
So I think we'd better check with `(uintptr_t)NULL`.